lib/Parser/Parse.cpp

```diff
@@ -1826,6 +1826,26 @@ void Parser::FinishParseBlock(ParseNodeBlock *pnodeBlock, bool needScanRCurly)
     }
 }
 
+void CheckFncExprNameCapturedInParamScope(ParseNodeFnc* pnodeFnc, ParseNodeBlock* pnodeFncExprScope)
```

Comment:
Rather than doing this as a separate walk of the ref stack, can it be folded into the binding pass? That's where we discover all the other stuff about a given symbol's references. If we know the thing we're binding is the name of a function expression, and if we see a reference in that function's param scope, then we can mark the scope accordingly. Would that work?

Reply:
Yeah, we can do that. I didn't because it's a little bit more messy than a separate walk. Probably worth doing, though. I cleaned it up a bit and pushed a commit which does this.